### PR TITLE
[node-manager] Declarative binding of SSHCredentials and StaticInstance

### PR DESCRIPTION
**Pull request by alexey-gavrilov-flant**
## Description
Object binding is redone to declarative binding of SSHCredentials and StaticInstance

## Why do we need it, and what problem does it solve?
This fix gives cleaner CAPS behavior/

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: chore
summary: Declarative binding of SSHCredentials and StaticInstance
impact_level: default 
```

### CODE REVIEW

**Comment by alexey-gavrilov-flant** [comment]
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Review by z9r5 — APPROVED** [review]

File: modules/040-node-manager/images/caps-controller-manager/src/api/deckhouse.io/v1alpha1/staticinstance_webhook.go [hunk 1]
```diff
@@ -77,11 +77,14 @@ func (r *StaticInstance) ValidateUpdate(old runtime.Object) (admission.Warnings,
...
+	if r.Status.CurrentStatus == nil ||
+		(r.Status.CurrentStatus.Phase != StaticInstanceStatusCurrentStatusPhaseError &&
+			r.Status.CurrentStatus.Phase != StaticInstanceStatusCurrentStatusPhaseInit &&
+			r.Status.CurrentStatus.Phase != StaticInstanceStatusCurrentStatusPhasePending) {
 		return nil, apierrors.NewForbidden(schema.GroupResource{
 			Group:    r.GroupVersionKind().Group,
 			Resource: "staticinstances",
-		}, r.Name, errors.New(`if you need to delete a StaticInstance that is not pending, you need to add the label '"node.deckhouse.io/allow-bootstrap": "false"' to the StaticInstance, after which you need to wait until the StaticInstance status becomes 'Pending'. Do not forget to decrease the 'NodeGroup.spec.staticInstances.count' field by 1, if needed`))
+		}, r.Name, errors.New(`if you need to delete a StaticInstance that is not pending or Error, you need to add the label '"node.deckhouse.io/allow-bootstrap": "false"' to the StaticInstance, after which you need to wait until the StaticInstance status becomes 'Pending'. Do not forget to decrease the 'NodeGroup.spec.staticInstances.count' field by 1, if needed`))
```
Comment (name212) [inline modules/040-node-manager/images/caps-controller-manager/src/api/deckhouse.io/v1alpha1/staticinstance_webhook.go:87]:
```suggestion
		}, r.Name, errors.New(`if you need to delete a StaticInstance that is not Pending or Error, you need to add the label '"node.deckhouse.io/allow-bootstrap": "false"' to the StaticInstance, after which you need to wait until the StaticInstance status becomes 'Pending'. Do not forget to decrease the 'NodeGroup.spec.staticInstances.count' field by 1, if needed`))
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Comment by alexey-gavrilov-flant** [comment]
/e2e/abort static;Static;containerd;1.27 10944221579 10944221579-1-con-1-27 /sys/deckhouse-oss/install:pr9369

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Comment by alexey-gavrilov-flant** [comment]
/e2e/abort static;Static;containerd;1.27 10944801013 10944801013-1-con-1-27 /sys/deckhouse-oss/install:pr9369 astra@192.168.199.38

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Comment by alexey-gavrilov-flant** [comment]
/e2e/abort static;Static;containerd;1.27 10960182024 10960182024-1-con-1-27 /sys/deckhouse-oss/install:pr9369 astra@192.168.199.152

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by alexey-gavrilov-flant** [comment]
/e2e/abort static;Static;containerd;1.27 10960823245 10960823245-1-con-1-27 /sys/deckhouse-oss/install:pr9369 astra@192.168.199.247

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @alexey-gavrilov-flant. I've started the workflow for label 'e2e/run/static'...

**Review by name212 — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Pull request merged**